### ISSUE TITLE
Ability to return custom responses?

### ISSUE THREAD

**Issue by oogali** [emersion/go-smtp#97]
I'd like to return custom responses in response to the execution of the `Mail`, `Rcpt`, and `Data` functions of the Session object.

I see I can string together my own responses using `SMTPError`, however the logic in the connection handling code doesn't particularly like receiving any sort of error.

Would it be against the spirit of the project if this logic was modified to further inspect SMTPError and only error out if SMTPError.Code was not 250?

**Comment by oogali** [comment]
As an example...

```
	return &smtp.SMTPError{
		Code:         250,
		EnhancedCode: smtp.EnhancedCode{2, 0, 0},
		Message:      fmt.Sprintf("Ok: queued as %s", queueID),
	}
```

**Comment by emersion** [comment]
We already allow this for a bunch of functions, see e.g.

https://github.com/emersion/go-smtp/blob/689d1d4f4d7c50a573182f27f390b9953d5de345/conn.go#L451

Maybe we're missing some others

**Comment by foxcpp** [comment]
> We already allow this for a bunch of functions, see e.g.

It is not going to work correctly since s.recipients list is not being populated.

**Comment by foxcpp** [comment]
Speaking of "spirit of the project". I believe having any useful text in success responses has marginal benefit since most (all?) client implementations will just discard it. What is the use case?

**Comment by foxcpp** [comment]
I you have fingerprinting concerns or just do not like silly responses, we can replace it with just "Ok". :)

**Comment by emersion** [comment]
Ah, right, this is about the success reply. Not sure it's worth it to do anything complicated to support this.

**Comment by oogali** [comment]
@foxcpp The use case is more for traceability -- to be able to trace the message from the ingestion point (go-smtp) to the final delivery/handoff.

This is more for troubleshooting and monitoring rather than for consumption by other SMTP clients.

**Comment by oogali** [comment]
I'm more interested in this on `DATA` than any other command (to surface the internal queue ID).

**Comment by gowthamgts** [comment]
@oogali - same use-case for me as well. @emersion - no plans to add this?

**Comment by kayrus** [comment]
@oogali @gowthamgts `Mail` and `Rcpt` should be solved in #188 
as for the `Data`, I'm trying to find a solution to get SMTP server response for a successful DATA command.
